crates/api/api_crud/src/post/update.rs

```diff
@@ -103,22 +103,43 @@ pub async fn update_post(
 
   check_community_user_action(&local_user_view, &orig_post.community, &mut context.pool()).await?;
 
-  if let Some(tags) = &data.tags {
-    // post view does not include communityview.post_tags
-    let community_view =
-      CommunityView::read(&mut context.pool(), orig_post.community.id, None, false).await?;
-    update_post_tags(
-      &context,
-      &orig_post.post,
-      &community_view,
-      tags,
-      &local_user_view,
-    )
-    .await?;
+  let is_post_creator = Post::is_post_creator(local_user_view.person.id, orig_post.post.creator_id);
+  let is_moderator = check_community_mod_action(
+    &local_user_view,
+    &orig_post.community,
+    false,
+    &mut context.pool(),
+  )
+  .await
+  .is_ok();
+
+  // both creator and moderator can change tags
+  if is_post_creator || is_moderator {
+    if let Some(tags) = &data.tags {
+      PostTag::update(&mut context.pool(), &orig_post.post, tags.clone()).await?;
+
+      // moderator cannot make any other changes, so return here
+      if is_moderator {
+        generate_post_link_metadata(
+          orig_post.post.clone(),
+          custom_thumbnail.flatten().map(Into::into),
+          |post| Some(SendActivityData::UpdatePost(post)),
+          context.clone(),
+        )
+        .await?;
+        return build_post_response(
+          context.deref(),
+          orig_post.community.id,
+          local_user_view,
+          post_id,
+        )
+        .await;
+      }
+    }
```

Comment:
This is to allow mods to edit post tags. Its rather ugly like this so it should probably be a separate endpoint like `PUT /api/v4/post/mod_edit`

Reply:
Ya I agree, much better to move it to its own endpoint and file. Much safer that way.